nixos/modules/system/boot/loader/grub/grub.nix

```diff
@@ -752,7 +752,7 @@ in
       # set at once.
       system.boot.loader.id = "grub";
 
-      environment.systemPackages = optional (grub != null) grub;
+      environment.systemPackages = mkIf (grub != null) [ grub ];
```

Comment:
In this instance it doesn't make a difference, on the other hand, it doesn't hurt either.

Reply:
It makes it easier to reason if we only used `mkIf` to set values in the module's config section.